examples/particle-swarm-optimizer.dx

```diff
@@ -0,0 +1,96 @@
+' # Particle Swarm Optimizer
+
+' ## Fitness function
+
+rosenbrock:: Real -> Real -> Real
+rosenbrock x y = sq (1.0 - x) + 10000.0*sq (y - x*x) 
+
+
+' We write one that uses vector for input
+
+rosenbrock2:: (2=>Real) -> Real
+rosenbrock2 xs =
+  x = xs.(asidx 0)
```

Comment:
Cases like this are really crying out for index literals. There's nothing remotely dynamic about this pair-to-2-vector conversion, so it's a shame to have to use the unchecked `asidx` (which should probably be called `unsafeAsIdx`). [We already have index literals in the abstract syntax](https://github.com/google-research/dex-lang/blob/master/src/lib/Syntax.hs#L56) we just haven't exposed them as concrete syntax. Perhaps we could try something like `0@3`, meaning "index 0 of the index set {0,1,2}".